src/main/java/org/spectrumauctions/sats/core/model/srvm/SRVMWorld.java

```diff
@@ -62,9 +63,12 @@ public int getNumberOfGoods() {
         return licenses;
     }
 
+    public ImmutableSet<SRVMBand> getBands() {
```

Comment:
I like the idea of caching instead of rebuildin the collection on each getter call, but can't we just return a Collections.umodifiableSet view? We could even store an unmodifiable view in the instance variable, as the list should be final. This would allow a simple getter and, at the same time, ensure that we don't by mistake modify the collection within our class (which would be possible with this PR)

Reply:
After various tries I actually fell back to the previous way to handle it, as you suggest - simply return a Collections.unmodifiableSet view.